docs/resources/source_table_kafka.md

```diff
@@ -0,0 +1,383 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "materialize_source_table_kafka Resource - terraform-provider-materialize"
+subcategory: ""
+description: |-
+  A Kafka source describes a Kafka cluster you want Materialize to read data from.
+---
+
+# materialize_source_table_kafka (Resource)
+
+A Kafka source describes a Kafka cluster you want Materialize to read data from.
+
+## Example Usage
+
+```terraform
+resource "materialize_source_table_kafka" "kafka_source_table" {
+  name          = "kafka_source_table"
+  schema_name   = "public"
+  database_name = "materialize"
+
+  source {
+    name          = materialize_source_kafka.test_source_kafka.name
+    schema_name   = materialize_source_kafka.test_source_kafka.schema_name
+    database_name = materialize_source_kafka.test_source_kafka.database_name
+  }
+
+  topic                   = "terraform"
+  include_key             = true
+  include_key_alias       = "message_key"
+  include_headers         = true
+  include_headers_alias   = "message_headers"
+  include_partition       = true
+  include_partition_alias = "message_partition"
+  include_offset          = true
+  include_offset_alias    = "message_offset"
+  include_timestamp       = true
+  include_timestamp_alias = "message_timestamp"
+
+
+  key_format {
+    text = true
+  }
+  value_format {
+    json = true
+  }
+
+  envelope {
+    upsert = true
+    upsert_options {
+      value_decoding_errors {
+        inline {
+          enabled = true
+          alias   = "decoding_error"
+        }
+      }
+    }
+  }
+
+  ownership_role = "mz_system"
+  comment        = "This is a test Kafka source table"
+}
+```
+
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Required
+
+- `name` (String) The identifier for the source table.
+- `source` (Block List, Min: 1, Max: 1) The source this table is created from. (see [below for nested schema](#nestedblock--source))
+- `topic` (String) The name of the Kafka topic in the Kafka cluster.
```

Comment:
this is actually no longer necessary to include for Kafka sources -- if the `(REFERENCE ..)` option in the statement is omitted it will still work since there is only one possible kafka topic that the table can reference